src/puppetlabs/trapperkeeper/bootstrap.clj

```diff
@@ -132,26 +136,102 @@
             (str "Unable to find bootstrap.cfg file via --bootstrap-config "
                  "command line argument, current working directory, or on classpath")))))
 
-(schema/defn chain-files :- [schema/Str]
-  "Takes a list of files, reads all their lines in, and returns a flattened seq
-   of all their lines put together.
+(schema/defn indexed
+  "Returns seq of [index, item] pairs
+  [:a :b :c] -> ([0 :a] [1 :b] [2 :c])"
+  [coll]
+  (map vector (iterate inc 0) coll))
 
-   In the case of an empty file, nil would be returned by line-seq, so we remove
-   all the nils at the end"
-  [files :- [(schema/protocol io/IOFactory)]]
-  (remove nil? (flatten (map #(line-seq (io/reader %)) files))))
+(schema/defn get-annotated-bootstrap-entries :- [AnnotatedBootstrapEntry]
+  [configs :- [(schema/protocol io/IOFactory)]]
+  (flatten (for [config configs]
+             (for [[line-number line-text] (indexed (map remove-comments (line-seq (io/reader config))))
+                   :when (not (empty? line-text))]
+               {:bootstrap-file (str config)
```

Comment:
This usage doesn't necessarily seem to match up with the schema for the fn.  The schema says that `config` can be any `IOFactory`, and I think that things like `InputStream` would satisfy that.  Here, you're assuming that the value has a meaningful `toString`, and storing it in a key called `bootstrap-file`, which implies that it's expected to be a file path.  Maybe, based on the usages of this function,  you could actually tighten up the schema to something like String or File?  In which case this line would be fine.